internal/provider/schemas/role.go

```diff
@@ -103,6 +103,36 @@ func ApiTokenRoleAttributeTypes() map[string]attr.Type {
 	}
 }
 
+func ResourceApiTokenRoleSchemaAttributes() map[string]resourceSchema.Attribute {
```

Comment:
if you look at `resource_deployment.go`'s `ValidateConfig`
We should have something like that for the roles.

For example, if its an organization token, then it can have any role
If it is a deployment token, it can just have one role
In this we should also check that the role matches the entity. Ie. workspace roles for workspace entities.

Reply:
we should also add a bunch of error test cases for these in the acceptance tests